runtime/lua/vim/loader.lua

```diff
@@ -337,14 +340,17 @@ function M.find(modname, opts)
   return results
 end
 
---- Resets the topmods cache for the path, or all the paths
+--- Resets the cache for the path, or all the paths
 --- if path is nil.
 ---@param path string? path to reset
 function M.reset(path)
   if path then
-    Loader._indexed[normalize(path)] = nil
+    path = normalize(path)
+    Loader._indexed[path] = nil
+    Loader._hashes[path] = nil
```

Comment:
This is not ideal, since `vim.loader.reset` expects a path on the rtp.

It's pretty much impossible to know all the lua mdules that would need to be cleared this way.

So either loop over all the hashes and clear those that start with the `path` parameter, or just always clear all hashes even when a `path` was given.

Reply:
For now, I've just cleared all the hashes. Can potentially improve this in the future.

Reply:
I think that's good enough. Not worth improving this I think